inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,249 @@
-    # ======
-    # Schema
-    # ======
-    if '$schema' in data and not data['$schema'].startswith('http'):
-        jsonschemas_ext = current_app.extensions.get('invenio-jsonschemas')
-        data['$schema'] = jsonschemas_ext.path_to_url(
-            "records/{0}".format(data['$schema']))
-
-    # ============================
-    # Collection
-    # ============================
-    data['collections'] = [{'primary': "HEP"}]
-    if form_fields['type_of_doc'] == 'thesis':
-        data['collections'].append({'primary': "THESIS"})
-
-    if get_value(form_fields, "arxiv_eprints.categories", None):
-        # Check if it was imported from arXiv
-        data['collections'].extend([{'primary': "arXiv"},
-                                    {'primary': "Citeable"}])
-        # Add arXiv as source
-        if data.get("abstracts"):
-            data['abstracts'][0]['source'] = 'arXiv'
-        if form_fields.get("arxiv_id"):
-            data['external_system_numbers'] = [{
-                'value': 'oai:arXiv.org:' + form_fields['arxiv_id'],
-                'institute': 'arXiv'
-            }]
-    if "publication_info" in data:
-        pub_keys = data['publication_info'][0].keys()
-
-        has_pub_info = all([
-            key in pub_keys for key in (
-                'year', 'journal_issue', 'journal_volume')])
-        has_page_or_artid = any([
-            key in pub_keys for key in ('page_start', 'page_end', 'artid')])
-
-        if has_pub_info and has_page_or_artid:
-            # NOTE: Only peer reviewed journals should have this collection
-            # we are adding it here but ideally should be manually added
-            # by a curator.
-            data['collections'].append({'primary': "Published"})
-            # Add Citeable collection if not present
-            collections = [x['primary'] for x in data['collections']]
-            if "Citeable" not in collections:
-                data['collections'].append({'primary': "Citeable"})
-    # ============================
-    # Title source and cleanup
-    # ============================
-    try:
-        # Clean up all extra spaces in title
-        data['titles'][0]['title'] = " ".join(
-            data['titles'][0]['title'].split()
+    builder = LiteratureBuilder(source='submitter')
+
+    for author in form_fields.get('authors', []):
+        builder.add_author(builder.make_author(
+            author['full_name'],
+            affiliations=force_force_list(author['affiliation']),
+        ))
+
+    for supervisor in form_fields.get('supervisors', []):
+        builder.add_author(builder.make_author(
+            supervisor['full_name'],
+            affiliations=force_force_list(supervisor['affiliation']),
+            supervisor=True
+        ))
+
+    builder.add_title(title=form_fields.get('title', []))
+
+    document_type = 'conference paper' if form_fields.get('conf_name') \
+        else form_fields.get('type_of_doc', [])
+
+    builder.add_if_non_empty(
```

Comment:
This does not look very nice, for example above using `builder.add_title` and here `builder.add_if_non_empty`, why not just do:


```python
if document_type:
    builder.add_document_type(document_type)
```

Or at the top were you call `filter_empty_elements` make sure that all the empty elements are filtered out.